pyccel/parser/semantic.py

```diff
@@ -812,7 +812,7 @@ def _handle_function(self, expr, func, args, **settings):
             func = func.cls_name
             if func in (CudaThreadIdx, CudaBlockDim, CudaBlockIdx, CudaGridDim):
                 if 'kernel' not in self.scope.decorators\
-                    or 'device' not in self.scope.decorators:
+                    and 'device' not in self.scope.decorators:
```

Comment:
can you add this to the description so we can keep track of it.

Reply:
Fixed.